gnovm/pkg/gnolang/preprocess.go

```diff
@@ -1006,6 +1006,23 @@ func Preprocess(store Store, ctx BlockNode, n Node) Node {
 								n.Args[1] = args1
 							}
 						}
+						// Another special case for append: adding untyped constants.
+						// They must be converted to the array type for consistency.
+						for i, arg := range n.Args[1:] {
+							if _, ok := arg.(*ConstExpr); !ok {
+								// Consider only constant expressions.
+								continue
+							}
+							if t1 := evalStaticTypeOf(store, last, arg); t1 != nil && !isUntyped(t1) {
+								// Consider only untyped values (including nil).
+								continue
+							}
+							// Get the array type from the first argument and convert to it.
+							s0 := evalStaticTypeOf(store, last, n.Args[0])
+							tx := constType(arg, s0.Elem())
+							var arg1 Expr = Call(tx, arg)
```

Comment:
Nit:

```suggestion
							arg1 := Call(tx, arg)
```

Even if it returns *CallExpr, it doesn't matter, no?